Improve behaviour of floated groups, and fix display issues with iframe blocks nested inside

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR cleans up some behaviour of the floated group block and the iframe block:

* It adds a `display: block` to all `iframe` elements so they behave consistently across AMP and non-AMP (`amp-frame` appears to be largely styled as `display: block`).
* Adds a minimum width to the iframe block when nested inside of a floated group block when its width is 100%; otherwise, it collapses. 
* Adds a maximum width to the iframe block, so it cannot exceed the available space when it has a `px` width set. 
* Makes floated group blocks stack on mobile; we've done this in Custom CSS on a couple sites already; it seems like the most logical default, as having the group block squished to 50% of the screen space would rarely make sense. 

Closes #1710; closes #1723.

### How to test the changes in this Pull Request:

1. Create a page and add a couple of floated group blocks, containing:
   * An iframe block with the default width (`100%) 
   * An iframe block with a higher fixed `px` width, like `500px`
   * Some text content -- a paragraph block or two, or a list. 
2. View on the front end; note that:
   * The floated group block + iframe block with the default width disappears.

![image](https://user-images.githubusercontent.com/177561/160903202-be4b050c-9e73-48fc-99f7-af0a053f4a6a.png)

   * The floated group block + iframe block with a wider fixed width is too wide when you start scaling down the browser.

![image](https://user-images.githubusercontent.com/177561/160903284-7b4d4838-dada-49d3-a47d-d74854a9b26d.png)

   * The floated group block + regular content seems cramped on smaller screens.

![image](https://user-images.githubusercontent.com/177561/160903328-6cc1f1c0-cfa2-4937-84c4-8abe243c68e9.png)

3. Apply the PR and run `npm run build`.
4. Confirm the following:
    * The floated group block + iframe block with the default width is now visible.

![image](https://user-images.githubusercontent.com/177561/160903542-3e885b29-2c5f-4afe-b8ec-905c73335003.png)

    * The floated group block + iframe block with a wider fixed width doesn't exceed the available space (it is smaller than your selected width, but I think that makes sense in this case -- if you need a wider iframe, it shouldn't be floated).

![image](https://user-images.githubusercontent.com/177561/160903589-5216198f-ec82-41cb-8d9e-10a3a26be4a4.png)

    * The floated group block stacks on mobile.

![image](https://user-images.githubusercontent.com/177561/160903619-ed9a902e-050d-4279-8b45-06d762767bf8.png)

5. Double-check the page with AMP disabled as well, to make sure there are no unexpected issues!

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (changes requested):
Looking great at mobile and desktop viewports (the stacking on mobile in particular looks much better), but at the tablet breakpoint I'm seeing all three test Group blocks break outside of the viewport:

<img width="805" alt="Screen Shot 2022-03-31 at 10 47 24 AM" src="https://user-images.githubusercontent.com/2230142/161108018-9f66495c-83d0-402d-ad64-7e394df4dbe3.png">

<img width="798" alt="Screen Shot 2022-03-31 at 10 47 33 AM" src="https://user-images.githubusercontent.com/2230142/161108030-db625f90-a51d-4ac6-b59c-1e6debcecc5c.png">

In the inspector, I'm seeing some negative margins on these that I think need to be zeroed out at the tablet breakpoint:

<img width="1559" alt="Screen Shot 2022-03-31 at 10 50 43 AM" src="https://user-images.githubusercontent.com/2230142/161108483-4b1786c1-fa40-4537-ad19-623d6590b032.png">

laurelfulford commented:
Thanks @dkoo! 

I've made a few adjustments:

* I've tweaked the styles that pull the aligned blocks out to the left and right -- when scaling down the browser window, they should still just touch the edge at some point, but no longer get cut off like they do now. 
* I added the same iframe width fallback to the editor -- the problem wasn't as bad there, the iframe was still visible, but still kind of weirdly narrow. The selector's a little funkier though -- since there's also a `max-width: 100%`, I can't use the same `style*='width: 100%'` without it also matching with the `max-width`, so it's more specific. 
* I noticed the image blocks weren't respecting the max-width in the editor, so I fixed that.

Just let me know if you have any questions about the above, or if you notice anything else amiss! :)

dkoo reviewed (changes requested):
Since you mentioned image blocks, I did some testing in floated Group blocks and found that if you set the width of the image to something wider than the Group's max width, you get a slightly strange result in the editor:

<img width="1245" alt="Screen Shot 2022-04-05 at 1 03 35 PM" src="https://user-images.githubusercontent.com/2230142/161830697-25cfb7cf-786f-42d2-8345-f43d77f61c93.png">

It looks fine on the front-end, though, with the image maxing out at the Group's max-width.

laurelfulford commented:
Thanks @dkoo!

I was actually testing image blocks on their own, since they can be aligned left and right without a group block. I did find a similar problem there, though, and tried to fix both that and the nested-in-group-one to the best of my abilities. The fix isn't responsive, but it's hard to get that 1:1 with the front-end -- I think this is about as close as I can get it at the moment (but do let me know if you spot anything glaring!).

dkoo reviewed (approved):
This fixes the overlap I saw earlier, and I didn't see any glaringly bad behavior in the editor on on the front-end, using images in groups or not in groups.

I did see that the selection outline for an image within a floated group looks a little off, but this seems to me like an extreme edge case and doesn't really affect the editing experience much, IMO.

laurelfulford commented:
Thanks for all the testing on this one, @dkoo!
